src/averageAges.js

```diff
@@ -55,7 +69,19 @@ function calculateWomenAverageAge(people, withChildren) {
  * @return {number}
  */
 function calculateAverageAgeDiff(people, onlyWithSon) {
-  // write code here
+  const children = onlyWithSon
```

Comment:
Consider simplifying the logic to avoid iterating over the people array twice. Here's a possible refactoring:
const children = people.filter((p) => {
  const hasMother = people.some((m) => m.name === p.mother);
  return onlyWithSon ? (p.sex === 'm' && hasMother) : hasMother;
});


This way, we iterate through people only once, and the condition is checked within the same iteration